include/elf_parser.hpp

```diff
@@ -0,0 +1,191 @@
+#pragma once
+#include <bitset>
+#include <err.h>
+#include <fcntl.h>
+#include <gelf.h>
+#include <libelf.h>
+#include <print>
+#include <stdlib.h>
+#include <string>
+#include <sysexits.h>
+#include <unistd.h>
+
+class ElfParser
+{
+  public:
+    ElfParser(char** t_file_ptr)
+      : m_file(t_file_ptr){};
+    void openElf();
+    void printEhdr();
+    void printPhdr();
+    void printShdr();
+    //void printSym();
+    void getSection(std::string section);
+    void closeElf();
+
+  private:
+    int m_elf_class{ 0 };
+    int m_fd, m_fbin;
+    size_t m_n, m_shstrndx;
+
+    char** m_file;
```

Comment:
I'd recommend replacing this with std::string so this object owns the string data.